test/integration/node-specific/abort_signal.test.ts

```diff
@@ -792,8 +793,9 @@ describe('AbortSignal support', () => {
     describe('if reauth throws', () => {
       beforeEach(() => {
         sinon.stub(ConnectionPool.prototype, 'reauthenticate').callsFake(async function () {
+          const error = new Error('Rejecting reauthenticate for testing');
```

Comment:
makes the stack trace point to the correct spot

Reply:
Does `throw new Error()` not construct a stack trace pointing to the correct spot?

Reply:
Concretely, I don't understand what the difference between these two snippets are in terms of the stacktrace:

```typescript
const error = new Error();
throw error;
```

```typescript
throw new Error();
```

Reply:
There's an await between the error being declared and thrown in the first example. So when I make the error sync at the top of the function the current stack trace can be captured from where the command originated. After the sleep the only stack context preserved starts at the sleep because there is no await where we call "pool.reauthenticate()"

Reply:
Ah, got it.  There isn't a difference between the two, but if there's an await between the error declaration and throwing it there is.  Thanks